### PR TITLE
Update Instagram provider to match Instagram api and work again.

### PR DESCRIPTION
**Pull request by jeremykenedy**
Update work out the box the API calls in `getUserByToken` were calling deprecated methods.

### CODE REVIEW

File: src/Instagram/Provider.php [hunk 1]
```diff
@@ -70,19 +74,21 @@ protected function getUserByToken($token)
...
+            'avatar'        => $user['profile_picture'] ?? null,
+            'media_count'   => $user['media_count'] ?? null,
         ]);
     }
 
     public function getAccessToken($code)
     {
         $response = $this->getHttpClient()->post($this->getTokenUrl(), [
-            RequestOptions::FORM_PARAMS => $this->getTokenFields($code),
+            'form_params' => $this->getTokenFields($code),
```
Comment (atymic) [inline src/Instagram/Provider.php:88]: Why was this changed?
https://docs.guzzlephp.org/en/stable/request-options.html#form-params

File: src/Instagram/Provider.php [hunk 2]
```diff
@@ -70,19 +74,21 @@ protected function getUserByToken($token)
...
         ]);
 
-        $this->credentialsResponseBody = json_decode((string) $response->getBody(), true);
```
Comment (atymic) [inline src/Instagram/Provider.php:85]: Same here, this is correct.

File: src/Instagram/Provider.php [hunk 3]
```diff
@@ -44,24 +53,19 @@ protected function getTokenUrl()
...
-        $response = $this->getHttpClient()->get(
-            'https://api.instagram.com/v1/users/self',
-            [
-                RequestOptions::QUERY   => $query,
-                RequestOptions::HEADERS => [
-                    'Accept' => 'application/json',
-                ],
-            ]
-        );
+        $meUrl = 'https://graph.instagram.com/me?access_token='.$token.'&fields='.implode(',', $this->fields);
```
Comment (atymic) [inline src/Instagram/Provider.php:56]: Can you link to the docs? this look a major change in API

File: src/Instagram/Provider.php [hunk 4]
```diff
@@ -2,23 +2,32 @@
...
      * {@inheritdoc}
      */
     protected $scopeSeparator = ' ';
 
+    /**
+     * The user fields being requested.
+     *
+     * @var array
+     */
+    protected $fields = ['account_type', 'id', 'username', 'media_count', 'profile_picture'];
```
Comment (atymic) [inline src/Instagram/Provider.php:25]: Do these need to be customisable?

**Review by atymic — CHANGES_REQUESTED** [review]

File: src/Instagram/Provider.php (hunk 3 above)
Comment (jeremykenedy) [reply src/Instagram/Provider.php:56]: https://developers.facebook.com/docs/instagram-basic-display-api/reference/me

https://developers.facebook.com/docs/instagram-basic-display-api/reference/user

https://developers.facebook.com/docs/instagram-basic-display-api/reference/oauth-authorize

File: src/Instagram/Provider.php (hunk 1 above)
Comment (jeremykenedy) [reply src/Instagram/Provider.php]: This was an oversite and has been reverted back.

File: src/Instagram/Provider.php (hunk 2 above)
Comment (jeremykenedy) [reply src/Instagram/Provider.php]: This was an oversite and has been reverted back.

File: src/Instagram/Provider.php (hunk 4 above)
Comment (jeremykenedy) [reply src/Instagram/Provider.php:25]: No, those are the only fields available. https://developers.facebook.com/docs/instagram-basic-display-api/reference/user#fields

File: src/Instagram/Provider.php (hunk 3 above)
Comment (jeremykenedy) [reply src/Instagram/Provider.php:56]: Meta is trying to discourage people from using login with IG and directing them to use FB. So the data is limited. the previous endpoint of `/users/self` no longer exists and will 404. I believe they replaced it with `/me`.

**Review by atymic — APPROVED** [review]
